cloud/parameterstore/ssm_client.go

```diff
@@ -0,0 +1,178 @@
+//nolint:unused
+package parameterstore
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmTypes "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	"github.com/evergreen-ci/utility"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws"
+)
+
+type ssmClient interface {
+	// PutParameter puts a parameter into Parameter Store.
+	PutParameter(context.Context, *ssm.PutParameterInput) (*ssm.PutParameterOutput, error)
+	// DeleteParametersSimple is the same as DeleteParameters but only returns
+	// the names of the deleted parameters rather than the full output.
+	DeleteParametersSimple(context.Context, *ssm.DeleteParametersInput) ([]string, error)
+	// DeleteParameters deletes the specified parameters. Parameter Store limits
+	// how many parameters can be deleted per call, so implementations are
+	// expected to handle batching transparently. The returned output slice contains the
+	// output of each batched request.
+	DeleteParameters(context.Context, *ssm.DeleteParametersInput) ([]*ssm.DeleteParametersOutput, error)
+	// GetParametersSimple is the same as GetParameters but only returns the
+	// parameter information rather than the full output.
+	GetParametersSimple(context.Context, *ssm.GetParametersInput) ([]ssmTypes.Parameter, error)
+	// GetParameters retrieves the specified parameters. Parameter Store
+	// limits how many parameters can be retrieved per call, so implementations
+	// must handle batching transparently. The returned output slice contains
+	// the output of each batched request.
+	GetParameters(context.Context, *ssm.GetParametersInput) ([]*ssm.GetParametersOutput, error)
+}
+
+type ssmClientImpl struct {
+	client *ssm.Client
+}
+
+// configCache is a mapping of configuration options to the cached AWS
+// configuration created by it. Currently the only supported option is the AWS
+// region.
+var configCache map[string]*aws.Config = make(map[string]*aws.Config)
+
+func newSSMClient(ctx context.Context, region string) (ssmClient, error) {
+	if region == "" {
+		region = "us-east-1"
```

Comment:
nit: evergreen.DefaultEC2Region

Reply:
This is a little wonky, but we can't actually depend on the `evergreen` package (or any evergreen repo package for that matter), because Parameter Store will eventually need to be used to store the admin settings, which would produce a cycle between the packages.

I guess we could move this into the top-level package, but it seems like it would clutter it up a lot.

Reply:
Oh wow, in that case it's fine as is.